src/Aspire.Hosting.Azure.WebPubSub/AzureWebPubSubExtensions.cs

```diff
@@ -56,6 +56,35 @@ public static IResourceBuilder<AzureWebPubSubResource> AddAzureWebPubSub(this ID
             construct.Add(new ProvisioningOutput("endpoint", typeof(string)) { Value = BicepFunction.Interpolate($"https://{service.HostName}") });
 
             construct.Add(service.CreateRoleAssignment(WebPubSubBuiltInRole.WebPubSubServiceOwner, construct.PrincipalTypeParameter, construct.PrincipalIdParameter));
+
+            var resource = (AzureWebPubSubResource)construct.Resource;
+            foreach (var setting in resource.Hubs)
+            {
+                var hubName = setting.Key;
+                
+                var hubBuilder = setting.Value;
+                var hubResource = hubBuilder;
+                var hub = new WebPubSubHub(Infrastructure.NormalizeIdentifierName(hubResource.Name))
+                {
+                    Name = setting.Key,
+                    Parent = service
+                };
+
+                // create the hub settings with default values
+                if (hub.Properties.Value == null)
+                {
+                    hub.Properties = new WebPubSubHubProperties();
+                }
+                // add to construct
+                construct.Add(hub);
+                // invoke the configure if any
+                hubResource.ConfigureConstruct(construct);
```

Comment:
With https://github.com/dotnet/aspire/pull/6258, we've gone away with these inline callbacks. Instead, the developer can `.ConfigureConstruct` on the parent object. That configure method gets the whole `Infrastructure` object, which contains all the bicep resources. The developer can tweak the bicep from there, even on child resources. For example, see 

https://github.com/dotnet/aspire/blob/4e7e0de000bb83f824cca2773892a22b5da89245/playground/cdk/CdkSample.AppHost/Program.cs#L64-L80

Reply:
I guess this is the same as what @davidfowl says above - https://github.com/dotnet/aspire/pull/5478#discussion_r1799759275